modular-framework/rwlock.h

```diff
@@ -0,0 +1,88 @@
+/* Copyright (c) 2017, ARM Limited. All rights reserved.
+ *
+ * Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef RWLOCK_H
+#define RWLOCK_H
+
+/* Get ride on ODP rwlock implementation and in meanwhile
```

Comment:
This is an example of why some code (atomic, barrier, cpu, hash, locks, time, thread, system, shared mem, random, arch/x86, arch/arm, ...) should exist in a util or base static library.  This way, other code can reuse it instead of duplicating or making wrapper layers for it.